plugins/trendmicro_visionone/icon_trendmicro_visionone/connection/connection.py

```diff
@@ -0,0 +1,66 @@
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import (
+    ConnectionTestException,
+    PluginException,
+)
+from .schema import ConnectionSchema, Input
+
+# Custom imports below
+import pytmv1
+
+
+class Connection(insightconnect_plugin_runtime.Connection):
+    def __init__(self):
+        super(self.__class__, self).__init__(input=ConnectionSchema())
+
+    def connect(self, params: dict = None):
+        """
+        Connection config params are supplied as a dict in
+        params or also accessible in self.parameters['key']
+
+        The following will setup the var to be accessed
+          self.blah = self.parameters['blah']
+        in the action and trigger files as:
+          blah = self.connection.blah
+        """
+        self.logger.info("Connect: Connecting...")
+
+        url = params.get(Input.API_URL, "")
+        if not url:
+            raise PluginException(
+                cause="No API URL provided.",
+                assistance="API URL is a required parameter, please provide a valid API URL.",
+                data=url,
+            )
+        key = params.get(Input.API_KEY).get("secretKey", "")
+        if not key or "default" in key:
+            raise PluginException(
+                cause="No API Key provided.",
+                assistance="API Key is a required parameter, please provide a valid API Key.",
+                data=key,
+            )
+        app = params.get(Input.APP_NAME, "Rapid7-InsightConnect")
+
+        self.server = url
+        self.token_ = key
+        self.app = app
+
```

Comment:
You shoudl establish the client within the connection, then just call it within the actions via `self.client.connection.method_name()`
```suggestion
self.client = pytmv1.client(app, key, url)
```

Reply:
Here is an example of an API call within an action from a plugin I worked on recently.
```
    def run(self, params={}):

        user_id = params.get(Input.ID)

        response = self.connection.client.get_user_by_id(user_id=user_id)

        return {Output.SUCCESS: response}
```